BookingProject/BookingWebAPI/BookingWebApi/Controllers/ApartamentController.cs

```diff
@@ -0,0 +1,37 @@
+﻿using BookingWebApi.Application.DTOs.ApartamentDTOs;
+using BookingWebApi.Application.Extensions;
+using BookingWebApi.Application.Filters;
+using BookingWebApi.Application.Interfaces;
+using BookingWebApi.Helpers;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Cryptography.X509Certificates;
+
+namespace BookingWebApi.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ApartamentController(
+        IApartamentService _apartamentService    
+    ) : ControllerBase
+    {
+        [Authorize(Roles ="Host")]
+        [HttpPost("CreateApartament")]
+        public async Task<IActionResult> CreateApartament(ApartamentPostDto apartamentDto)
+        {
+            var result = await _apartamentService.CreateApartament(apartamentDto, UserHelpers.GetUserId(HttpContext));
+
+            return result.ToResponse();
+        }
+
+        
+        [HttpGet("GetApartaments")]
```

Comment:
This is not REST convention - you should use
POST api/apartaments to create
GET api/apartaments?<query-params> to get it.

Also rename ur controller to be Apartament**s**

Reply:
Fixed